recommender/static/js/src/recommender.js

```diff
@@ -1,1022 +1,1187 @@
-						backToView();
-					}
-				});
-			});
-		});
-	}
-
-    /* Add tooltips to each global component */
-	function addTooltip() {
-		tooltipsCats.forEach(function(cats, ind) {
-			var classes = cats.split(".");
-			try {
-				$("." + classes[1], element).tooltipster('destroy');
-			}
-			catch (e) {  }
-		});
-		tooltipsCats.forEach(function(cats, ind) {
-			var classes = cats.split(".");
-			try {
-				if (classes.length == 3 && (! $("." + classes[1], element).hasClass(classes[2]) )) {
-					$("." + classes[1], element).tooltipster({
-						content: $('<span>' + tooltipsCatsText["." + classes[1]] + '</span>'),
-						theme: '.my-custom-theme',
-						maxWidth: '300'
-					});
-					return;
-				}
-				if ($(cats, element).hasClass('tooltipstered')) { return; }
-				$(cats, element).tooltipster({
-					content: $('<span>' + tooltipsCatsText[cats] + '</span>'),
-					theme: '.my-custom-theme',
-					maxWidth: '300'
-				}); 
-			}
-			catch (e) {  }
-		});
- 	}
-
-    /* Add tooltips to each cat in cats */
-	function addTooltipPerCats(cats) {
-		cats.forEach(function(cat, ind) {
-			try {
-				$(cat, element).tooltipster('destroy');
-			}
-			catch (e) {  }
-		});
-		cats.forEach(function(cat, ind) {
-			try {
-				$(cat, element).tooltipster({
-					content: $('<span>' + tooltipsCatsText[cat] + '</span>'),
-					theme: '.my-custom-theme',
-					maxWidth: '300'
-				}); 
-			}
-			catch (e) {  }
-		});
- 	}
-
-	/* Add tooltips to each component in each resource */
-	function addTooltipPerResource(ele) {
-        tooltipsCatsPerResource.forEach(function(cats, ind) {
-			var classes = cats.split(".");
-            if (classes.length == 3) {
-                try {
-                    $(ele, element).find("." + classes[1]).tooltipster('destroy');
+            $('.editTitle,.editUrl,.editDescriptionText', element).unbind();
+            $('.editTitle,.editUrl,.editDescriptionText', element).bind('input propertychange', function() { enableEditSubmit(); });
+            $('.editResourceForm', element).find("input[name='file']").unbind();
+            $('.editResourceForm', element).find("input[name='file']").change(function() {
+                if ($(this).val() !== '') { enableEditSubmit(); }
+            });
+            
+            /* Add tooltips for editting page */
+            addTooltipPerCats(tooltipsEditCats);
+
+            /* Upload the screenshot, edit the resource in the database, and update the resource list */
+            $('.editSubmit', element).unbind();
+            $('.editSubmit', element).click(function() {
+                /* data: resource to be submitted to database */
+                data['url'] = $('.editUrl', element).val();
+                data['title'] = $('.editTitle', element).val();
+                data['descriptionText'] = $('.editDescriptionText', element).val();
+                data['description'] = ''
+                if (data['url'] === '' || data['title'] === '') { return; }
```

Comment:
Nitpick: might be simpler to do
```js
if (!data.url || !data.title) {
  return;
}
```
That will return early if the `url` or `title` params are unset, undefined, 0, the empty string, or any other falsy value in Javascript -- which is *probably* what you want, but you should confirm that.